### PR TITLE
raptor train

### PR DESCRIPTION
**Pull request by spacek531**
Making the pull request now. Having some problems getting OpenRCT2 to recognize the object

### CODE REVIEW

**Comment by spacek531** [comment]
I created a junction from `openrct2/objects` to the folder and OpenRCT2 does not load the object. Using `OpenRCT2.com` does not provide any console logs regarding the object. Not sure what's going on.

File: objects/official/ride/official.rmct2/object.json [hunk 1]
```diff
@@ -0,0 +1,3470 @@
+{
+    "id": "official.rmct2",
+    "authors": ["spacek531"],
+    "version": "1.0",
+    "originalId": "00000000|RMCT2   |00003000",
+    "objectType": "ride",
+    "properties": {
+        "type": 93,
```
Comment (duncanspumpkin) [inline objects/official/ride/official.rmct2/object.json:8]: set the type as `"single_rail_rc"`

**Comment by duncanspumpkin** [comment]
Can you upload the images as well.

File: objects/official/ride/official.rmct2/object.json [hunk 2]
```diff
@@ -0,0 +1,3470 @@
...
+                "VEHICLE_ENTRY_FLAG_RIDERS_SCREAM": true
+            }
+        ],
+        "loadingPositions": [
+            [-2],
+            [0]
+        ]
+    },
+    "images": [
+		{ "path": "images\preview.png", "x": 0, "y": 0 },
```
Comment (Gymnasiast) [inline objects/official/ride/official.rmct2/object.json:95]: Should be a forward slash (/).

File: objects/official/ride/official.rmct2/object.json [hunk 3]
```diff
@@ -0,0 +1,3470 @@
+{
+    "id": "official.rmct2",
+    "authors": ["spacek531"],
+    "version": "1.0",
+    "originalId": "00000000|RMCT2   |00003000",
```
Comment (Gymnasiast) [inline objects/official/ride/official.rmct2/object.json:5]: This is not correct. It should consist of the DAT header and checksum of the original .DAT object if there is one. If not, let me know and I´ll calculate the values.

File: objects/official/ride/official.rmct2/object.json [hunk 4]
```diff
@@ -0,0 +1,3470 @@
...
+    "objectType": "ride",
+    "properties": {
+        "type": 93,
+        "category": "rollercoaster",
+        "separateRide": true,
```
Comment (Gymnasiast) [inline objects/official/ride/official.rmct2/object.json:10]: This property is no longer used, so just leave this out.

**Comment by Gymnasiast** [comment]
I´ll help you get the object working. I have already added some line notes of stuff to address.

Is this supposed to be an OpenRCT2-only object, or do you intend to make it available for RCT2?

**Comment by duncanspumpkin** [comment]
> Is this supposed to be an OpenRCT2-only object, or do you intend to make it available for RCT2?

Its for the new coaster type so it would never load in RCT2.

File: objects/official/ride/official.rmct2/object.json (hunk 3 above)
Comment (spacek531) [reply objects/official/ride/official.rmct2/object.json:5]: This is the actual header of the real DAT

**Comment by spacek531** [comment]
I have excluded the images thus far because they are quite large, totalling about 3.5 Mb. I'm told I can compress the images, but I do not know how to do this. If you would like me to upload the images as-is, I can do so.

**Comment by Gymnasiast** [comment]
> Its for the new coaster type so it would never load in RCT2.

@duncanspumpkin It is still possible to have the corresponding DAT file use a different ride type, like X7 did with the RMC trains (which are Twister RC trains in RCT2). Hence the question if @spacek531 was planning to make it available for RCT2 or not.

**Comment by spacek531** [comment]
>Hence the question if @spacek531 was planning to make it available for RCT2 or not.

I am not

**Comment by Gymnasiast** [comment]
OK. Can you zip up the images and drag the ZIP into a comment here? I can take care of compressing them.

**Comment by spacek531** [comment]
[images.zip](https://github.com/OpenRCT2/objects/files/5729283/images.zip)

**Comment by spacek531** [comment]
image

I've tested the object with the new images, and everything works fine. There's a strange freezing going on when hovering over the object in the object selection window, which I will file an issue for in the OpenRCT2 project.

**Review by duncanspumpkin — APPROVED** [review]

**Comment by IntelOrca** [comment]
We think using less .png files for the object will help reduce the load times. I.e. storing multiple sprites in the same .png file and referencing their x,y,w,h in the json file. Is it worth doing this before merging?

**Comment by duncanspumpkin** [comment]
Thought we were going to investigate that for later.

**Comment by IntelOrca** [comment]
@duncanspumpkin I know but isn't merging a lot of binary files a point of no return in terms of repository size?

**Comment by IntelOrca** [comment]
@spacek531 I would like to try using one (or few .pngs) containing all the sprites. Are you able to generate a single .png file containing them all and JSON where it specifies `srcX`, `srcY`, `width` and `height` for each sprite?

**Comment by spacek531** [comment]
I will see what I can do

**Comment by Fred-1044276** [comment]
Is there any way you can reduce the train object file size such that it doesn't cause lag/frame rate drops whenever you hover over it in the object selection window?

**Comment by duncanspumpkin** [comment]
@Fred-1044276 that's what we are trying to do and why this hasn't been merged

**Comment by AaronVanGeffen** [comment]
Could it be a solution to embed the images as a packed (RLE-encoded) object, in a manner similar to g2.dat? I realise that defeats the purpose of an easily editable object. However, if the packing is done at release time, that still leaves an easily editable object in the repository.

**Comment by spacek531** [comment]
> @spacek531 I would like to try using one (or few .pngs) containing all the sprites. Are you able to generate a single .png file containing them all and JSON where it specifies `srcX`, `srcY`, `width` and `height` for each sprite?

Is OpenRCT2 currently capable of handling this? I'm looking to finalize this soon.

**Comment by duncanspumpkin** [comment]
I think Ted has added that capability. @IntelOrca could you confirm

**Comment by IntelOrca** [comment]
On NSF, but could be back ported

**Comment by karst** [comment]
I was wondering if it might be an idea to have this (or any of the other two spacek made raptor trains) as additional versions/skins of official object?

**Comment by oli414** [comment]
Superseded by #166

**Pull request closed without merging**